python/ml4ir/base/model/scoring/Monte_Carlo_Scorer.py

```diff
@@ -0,0 +1,131 @@
+    """
+
+    def __init__(
+            self,
+            model_config: dict,
+            feature_config: FeatureConfig,
+            interaction_model: InteractionModel,
+            loss: RelevanceLossBase,
+            file_io: FileIO,
+            aux_loss: Optional[RelevanceLossBase] = None,
+            aux_loss_weight: float = 0.0,
+            aux_metrics: Optional[List[Union[Metric, str]]] = None,
+            output_name: str = "score",
+            logger: Optional[Logger] = None,
+            logs_dir: Optional[str] = "",
+            **kwargs
+    ):
+        """
+        Constructor method for creating a RelevanceScorer object
+
+        Parameters
+        ----------
+        model_config : dict
+            Dictionary defining the model layer configuration
+        feature_config : `FeatureConfig` object
+            FeatureConfig object defining the features and their configurations
+        interaction_model : `InteractionModel` object
+            InteractionModel that defines the feature transformation layers
+            on the input model features
+        loss : `RelevanceLossBase` object
+            Relevance loss object that defines the final activation layer
+            and the loss function for the model
+        file_io : `FileIO` object
+            FileIO object that handles read and write
+        aux_loss : `RelevanceLossBase` object
+            Auxiliary loss to be used in conjunction with the primary loss
+        aux_loss_weight: float
+            Floating point number in [0, 1] to indicate the proportion of the auxiliary loss
+            in the total final loss value computed using a linear combination
+            total loss = (1 - aux_loss_weight) * loss + aux_loss_weight * aux_loss
+        aux_metrics: List of keras.metrics.Metric
+            Keras metric list to be computed on the aux label
+        output_name : str, optional
+            Name of the output that captures the score computed by the model
+        logger : Logger, optional
+            Logging handler
+        logs_dir : str, optional
+            Path to the logging directory
+
+        Notes
+        -----
+        logs_dir : Used to point model architectures to local logging directory,
+            primarily for saving visualizations.
+        """
+        super().__init__(feature_config=feature_config,
+                         model_config=model_config,
+                         interaction_model=interaction_model,
+                         loss=loss,
+                         aux_loss=aux_loss,
+                         aux_loss_weight=aux_loss_weight,
+                         aux_metrics=aux_metrics,
+                         output_name=output_name,
+                         logger=logger,
+                         file_io=file_io,
+                         logs_dir=logs_dir,
+                         **kwargs)
+        self.monte_carlo_inference_trials = self.model_config["monte_carlo_inference_trials"]["value"]
```

Comment:
@mohazahran Can you change value to `num_trials`? You can modify main key to `monte_carlo_inference` to keep it generic for future in case.

Also, move them to `keys.py` maybe and use constants here instead of strings?